test/functional/mempool_limit.py

```diff
@@ -20,55 +23,52 @@ def set_test_params(self):
-        self.skip_if_no_wallet()
-
     def run_test(self):
-        txouts = gen_return_txouts()
-        relayfee = self.nodes[0].getnetworkinfo()['relayfee']
+        node = self.nodes[0]
+        miniwallet = MiniWallet(node)
+        relayfee = node.getnetworkinfo()['relayfee']
 
-        self.log.info('Check that mempoolminfee is minrelytxfee')
-        assert_equal(self.nodes[0].getmempoolinfo()['minrelaytxfee'], Decimal('0.00001000'))
-        assert_equal(self.nodes[0].getmempoolinfo()['mempoolminfee'], Decimal('0.00001000'))
+        self.log.info('Check that mempoolminfee is minrelaytxfee')
+        assert_equal(node.getmempoolinfo()['minrelaytxfee'], Decimal('0.00001000'))
+        assert_equal(node.getmempoolinfo()['mempoolminfee'], Decimal('0.00001000'))
 
-        txids = []
-        utxos = create_confirmed_utxos(self, relayfee, self.nodes[0], 91)
+        # Generate UTXOs to flood the mempool
+        # 1 to create a tx initially that will be evicted from the mempool later
+        # 3 batches of multiple transactions with a fee rate much higher than the previous UTXO
+        # And 1 more to verify that this tx does not get added to the mempool with a fee rate less than the mempoolminfee
+        self.generate(miniwallet,1 + (3 * 25) + 1)
+
+        # Mine 99 blocks so that the UTXOs are allowed to be spent
+        self.generate(node,COINBASE_MATURITY - 1)
 
         self.log.info('Create a mempool tx that will be evicted')
-        us0 = utxos.pop()
-        inputs = [{ "txid" : us0["txid"], "vout" : us0["vout"]}]
-        outputs = {self.nodes[0].getnewaddress() : 0.0001}
-        tx = self.nodes[0].createrawtransaction(inputs, outputs)
-        self.nodes[0].settxfee(relayfee) # specifically fund this tx with low fee
-        txF = self.nodes[0].fundrawtransaction(tx)
-        self.nodes[0].settxfee(0) # return to automatic fee selection
-        txFS = self.nodes[0].signrawtransactionwithwallet(txF['hex'])
-        txid = self.nodes[0].sendrawtransaction(txFS['hex'])
-
-        relayfee = self.nodes[0].getnetworkinfo()['relayfee']
-        base_fee = relayfee*100
-        for i in range (3):
-            txids.append([])
-            txids[i] = create_lots_of_big_transactions(self.nodes[0], txouts, utxos[30*i:30*i+30], 30, (i+1)*base_fee)
+        tx_to_be_evicted_id = miniwallet.send_self_transfer(from_node=node, fee_rate=relayfee)["txid"]
+
+        # Increase the tx fee rate massively to give the subsequent transactions a higher priority in the mempool
+        base_fee = relayfee * 1000
+
+        self.log.info('Fill up the mempool with txs with higher fee rate')
+        no_of_large_tx_sent = 0
+        txouts = gen_return_txouts()
+
+        for batch_of_txid in range(1, 4):
+            # Increment the tx fee rate gradually by a factor of (base_fee) for each batch
+            no_of_large_tx_sent += miniwallet.send_large_transactions(node, txouts, 25, batch_of_txid * base_fee)
```

Comment:
in https://github.com/bitcoin/bitcoin/pull/22543/commits/4902ce86afc6b9c8a7a72b1c91466fe26a00f9b9:

id suggest using a named variable (instead of just 25) to make this more clear. also, `no_of_large...` should be `num_of_large...`:

```suggestion
tx_batch_size = 25
for batch_of_txid in range(1, 4):
    # Increment the tx fee rate gradually by a factor of (base_fee) for each batch
    miniwallet.send_large_transactions(node, txouts, tx_batch_size, batch_of_txid * base_fee) 
    num_of_large_tx_sent += tx_batch_size

Reply:
Thank you for the naming suggestion. Has been fixed